source/base/parameter_handler.cc

```diff
@@ -736,6 +736,32 @@ ParameterHandler::parse_input_from_json(std::istream &in,
 
 
 
+void
+ParameterHandler::create_default_input_file(const std::string filename,
```

Comment:
How do you intend to handle the [other output styles](https://www.dealii.org/developer/doxygen/deal.II/classParameterHandler.html#a8364dda711b93753c6809eefe2a8e827)? In particular, `ShortText`, `ShortXML`, `ShortJSON`.

Reply:
Good question. Currently I ignored that because ``ParameterAcceptor`` did also not do that and I first wanted to stay close to the original implementation. Of course, one should think about this. My opinion is that in terms of design principles the current design with all the ``Short...`` versions is a violation of the open-closed principle, so I would vote for a boolean ``short_version`` and entities of ``OutputStyle`` should be "unique" identifiers.

Reply:
I included the Short versions now. 

Independently from this PR, the design would be improved in my opinion by passing ``short_version`` also to ``print_parameters()`` and removing ``OutputStyle::ShortText/JSON/XML``. I briefly looked at the implementation and things would really become simpler by this change.

Reply:
We had this discussion a couple of weeks ago in https://github.com/dealii/dealii/pull/9707#pullrequestreview-379073429. I like the suggestion made by @bangerth there because it resembles the approach also used in `FEValues` to specify the quantities to be updated. But I did not find any time to implement it then.

Reply:
At many places in the code we have if-statements like ``if(output_style == OutputStyle::Text || output_style == OutputStyle::ShortText)``.  For me, the design problem is that ``ShortText`` etc. is an entity of ``OutputStyle``.

Reply:
I am afraid that the enum entries `Text`, `LaTeX`, `Description`, `XML`, `JSON`, and `ShortText` have stay part of `ParameterHandler::OutputStyle` for compatibility reasons (I have not listed `ShortXML` and `ShortJSON` since they have been added only recently since the last release). The only way out what I see (as suggested by @bangerth) would be introduce new entries in the enum and redefine existing entries, like:

```cpp
  enum OutputStyle {
    with_documentation = 0,                    //  0
    without_documentation = 1,                 //  1

   Text = 2,                                   //  2
   ShortText = Text | without_documentation,   //  3

  LaTex = 4,                                   //  4

  Description = 8,                             //  8

  JSON = 16,                                   // 16
  ShortXML = XML | without_documentation,      // 17

  JSON = 32,                                   // 32
  ShortJSON = JSON | without_documentation,    // 33
  }
```

Reply:
It's ok. I can live with it if the short versions will remain part of OutputStyle. We should just be aware of that and not force new functions and interfaces to adhere to that redundancy in design.